BHoM_Engine/Compute/RecordEvent.cs

```diff
@@ -46,6 +46,31 @@ public static bool RecordEvent(string message, EventType type = EventType.Unknow
 
         /***************************************************/
 
+        [Description("Record an event with details of a C# exception within the BHoM logging system.")]
+        [Input("exception", "The C# exception being caught to provide the event and stack information for.")]
+        [Input("message", "An optional additional message which will be displayed first in the event log.")]
+        [Output("success", "True if the event has been successfully recorded as a BHoM Event.")]
+        public static bool RecordEvent(Exception exception, string message = "", EventType type = EventType.Unknown)
+        {
+            string exceptionMessage = "";
+
+            Exception e = exception;
+            while(e != null)
+            {
+                exceptionMessage += $"{e.Message}{Environment.NewLine}{Environment.NewLine}";
```

Comment:
@IsakNaslundBh already did a proper review of the PR, so only a minor code comment from me. From what I understand, `exceptionMessage` will always end with 2 line breaks. This may not look ideal neither in logs nor in the UI, so I would consider trimming the string at some point or slightly rewriting the code to something around below:

```suggestion
                if (!string.IsNullOrWhiteSpace(exceptionMessage ))
                    exceptionMessage += $"{Environment.NewLine}{Environment.NewLine}";
                
                exceptionMessage += e.Message;
```

This comment has been written without running the code, so please ignore if I missed a point 🙈

Reply:
Makes sense, implemented.